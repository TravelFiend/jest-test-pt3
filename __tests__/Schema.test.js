const Schema = require('../lib/Schema');

describe('Schema', () => {
    it('validates dat schema properly', () => {
        const schema = new Schema({
            brand: {
                type: String,
                required: true
            },
            spokes: {
                type: Number,
                required: true
            },
            saddle: {
                type: String
            }
        });

        const bike = {
            brand: 'Rocky Mountain',
            spokes: 32,
            saddle: 'leather'
        };

        expect(schema.validate(bike)).toEqual({
            brand: 'Rocky Mountain',
            spokes: 32,
            saddle: 'leather'
        });
    });

    it('should throw this weak sauce schema', () => {
        const schema = new Schema({
            brand: {
                type: String,
                required: true
            },
            spokes: {
                type: Number,
                required: true
            },
            saddle: {
                type: String
            }
        });

        const bike = {
            brand: 'Rocky Mountain',
            saddle: 'leather'
        };

        expect(() => schema.validate(bike)).toThrowErrorMatchingSnapshot();
    });
});
