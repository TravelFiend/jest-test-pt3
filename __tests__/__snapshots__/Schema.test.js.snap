// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Schema should throw this weak sauce schema 1`] = `"invalid schema >> Error: You're missing the spokes field"`;
